android/java/org/chromium/chrome/browser/toolbar/top/BraveToolbarLayoutImpl.java

```diff
@@ -590,6 +632,37 @@ public void didSelectTab(Tab tab, @TabSelectionType int type, int lastId) {
                 };
     }
 
+    private void showYouTubePipIcon(@NonNull final Tab tab) {
+        // The layout could be null in Custom Tabs layout.
+        if (mYouTubePipLayout == null) {
+            return;
+        }
+
+        // Hide the layout if the current tab is in a state where it doesn't support or allow PiP
+        // mode. This can also happen when a tab is re-selected after a crash and it's showing
+        // the crash custom view, or is in a frozen state (likely inactive or unloaded).
+        if (!BraveYouTubeScriptInjectorNativeHelper.isPictureInPictureAvailable(
+                tab.getWebContents())) {
+            mYouTubePipLayout.setVisibility(View.GONE);
+            return;
+        }
+        final boolean show =
+                PictureInPicture.isEnabled(getContext())
```

Comment:
shouldn't we check this first and return early?

Reply:
I see, not return early the, but by assigning both this and `BraveYouTubeScriptInjectorNativeHelper.isPictureInPictureAvailable` to vars you are always calling them both and we probably want to avoid making the jni call if `PictureInPicture.isEnabled` is not enabled. So `final boolean show = PictureInPicture.isEnabled(getContext()) && BraveYouTubeScriptInjectorNativeHelper.isPictureInPictureAvailable(
                        tab.getWebContents());`

This is not a critical change, but just fyi

Reply:
although thinking about it more, those considerations are less important than consolidating the logic so ideally we'd probably pass a `bool` to `BraveYouTubeScriptInjectorNativeHelper.isPictureInPictureAvailable` for `PictureInPicture.isEnabled(getContext())` instead of checking directly in `showYouTubePipIcon`. The current logic is very difficult to test effectively because it's distributed across multiple methods, classes and languages